### PR TITLE
Feat/#53 add users static

### PR DESCRIPTION
**Pull request by ciaranschutte**
Adds the static UI for add user modal

### CODE REVIEW

File: client/components/pages/submissionSystem/modals/addUser/formSchema.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import * as yup from 'yup';
+
+const testGsuite = async email => {
```
Comment (joneubank) [inline client/components/pages/submissionSystem/modals/addUser/formSchema.js]: Its a good placeholder... have you figured out how to actually confirm this yet?

File: client/components/pages/submissionSystem/modals/addUser/formSchema.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  return true;
+};
+
+const AddUserSchema = yup.object().shape({
+  firstName: yup.string().required('Please fill in first name'),
+  lastName: yup.string().required('Please fill in last name'),
+  email: yup
+    .string()
+    .required('Please fill in email')
+    .test('is-gsuite', 'email is not gsuite', testGsuite),
```
Comment (joneubank) [inline client/components/pages/submissionSystem/modals/addUser/formSchema.js]: A note about ordering the validations: 
We want to validate the form with the `abortEarly:false` option so that we find all the failures, not just the first. When this is done, the message that will be returned is for the each field will be from the last validation in the list that failed. In this case, if email is blank, it will fail the required validation, and the is-gsuite validation, and the messge returned would be "email is not gsuite". This would look weird since the email field is empty.

TL;DR - Put required as the last validation in the list.

File: client/components/pages/submissionSystem/modals/addUser/index.js [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+import Button from 'uikit/Button';
+import Icon from 'uikit/Icon';
+import Typography from 'uikit/Typography';
+import css from '@emotion/css';
+import UserSection from './userSection';
+import AddUserSchema from './formSchema';
+import * as yup from 'yup';
+import { get } from 'lodash';
+
+const useFormHook = ({ initialFields, schema: formSchema }) => {
```
Comment (joneubank) [inline client/components/pages/submissionSystem/modals/addUser/index.js]: longterm we can abstract this to use in both this form and create-program. this has all the boilerplate state management i wrote inline in the form component.

File: client/components/pages/submissionSystem/modals/addUser/formSchema.js (hunk 1 above)
Comment (ciaranschutte) [reply client/components/pages/submissionSystem/modals/addUser/formSchema.js]: https://app.zenhub.com/workspaces/argo-5c8c8637777e5f767e486217/issues/icgc-argo/argo-platform/50

apparently you can check against the DNS record... the specifics of this I do not know yet

File: client/components/pages/submissionSystem/modals/addUser/index.js (hunk 3 above)
Comment (ciaranschutte) [reply client/components/pages/submissionSystem/modals/addUser/index.js]: Sounds good. This is still a little too rigid. for instance it assumes each "section" is of the type "formSchema"

File: client/components/pages/submissionSystem/modals/addUser/formSchema.js (hunk 2 above)
Comment (ciaranschutte) [reply client/components/pages/submissionSystem/modals/addUser/formSchema.js]: great! nice catch

Comment (joneubank) [reply client/components/pages/submissionSystem/modals/addUser/formSchema.js]: Also if you provide a `.label("Email")` you can skip the error message for required... the default text is good enough

File: client/uikit/form/Input/index.js [hunk 4]
```diff
@@ -43,7 +44,10 @@ const Input = ({
       <StyledInputWrapper
         size={size}
         onFocus={() => setActive('focus')}
-        onBlur={() => setActive('default')}
+        onBlur={() => {
+          setActive('default');
+          onBlur();
```
[the remark this answers is not in the document]
Comment (ciaranschutte) [reply client/uikit/form/Input/index.js]: covered in Jon's merged PR :)

File: client/uikit/Icon/icons/index.js [hunk 5]
```diff
@@ -18,6 +18,7 @@ import warning from './collection/warning';
 import mail from './collection/mail';
 import trash from './collection/trash';
 import google from './collection/google';
+import plus_circle from './collection/plus_circle';
```
Comment (d8660091) [inline client/uikit/Icon/icons/index.js]: Do we have any name convention for js files?

File: client/components/pages/submissionSystem/modals/addUser/index.js [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+  const [form, setForm] = useState({ errors: [initialFields], data: [initialFields] });
+  const [touched, setTouched] = useState(false);
+  const { errors, data } = form;
+
+  const hasErrors = errors
+    .map(section => Object.values(section))
+    .flat()
+    .reduce((acc, val) => acc || !!val, false);
```
Comment (d8660091) [inline client/components/pages/submissionSystem/modals/addUser/index.js]: Array.prototype.some() is an alternative

File: client/components/pages/submissionSystem/modals/addUser/index.js [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+
+  // set form data
+  const setData = ({ key, val, index }) => {
+    if (!touched) setTouched(true);
+
+    setForm({
+      ...form,
+      data: data.map((field, i) => (i === index ? { ...field, [key]: val } : field)),
```
Comment (d8660091) [inline client/components/pages/submissionSystem/modals/addUser/index.js]: An alternative: _.set(data, `[${index}].${key}`, val)

File: client/uikit/Icon/icons/index.js (hunk 5 above)
Comment (ciaranschutte) [reply client/uikit/Icon/icons/index.js]: I just continued what was there already: use an underscore to split words

File: client/components/pages/submissionSystem/modals/addUser/userSection.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+  return (
+    <Section>
+      <div
+        css={css`
+          flex: 1;
+        `}
+      >
+        <Row nogutter>
+          <Col sm={6} style={{ paddingRight: '5px' }}>
```
Comment (hlminh2000) [inline client/components/pages/submissionSystem/modals/addUser/userSection.js]: these paddings are causes a slight shift in the layout on smaller screen:
Screen Shot 2019-07-11 at 3 57 44 PM

Comment (ciaranschutte) [reply client/components/pages/submissionSystem/modals/addUser/userSection.js]: oof, good catch thank you

File: client/components/pages/submissionSystem/modals/addUser/index.js (hunk 7 above)
Comment (ciaranschutte) [reply client/components/pages/submissionSystem/modals/addUser/index.js]: I think _.set returns same object and we need it immutable for setting new state

Comment (d8660091) [reply client/components/pages/submissionSystem/modals/addUser/index.js]: how about _.set({...data})?
